api/v1alpha1/gateway_types.go

```diff
@@ -54,78 +57,200 @@ type GatewayList struct {
+}
+
+// Listener embodies the concept of a logical endpoint where a
+// Gateway can accept network connections.
+//
+// The GatewayClass may collapse Listener definitions into single
+// implementation-defined acceptor configuration. This allows Gateways to
+// be implemented in terms of virtual services configured within the same
+// proxy server process.
+//
+// Listeners may be considered collapsible if all of the following
+// conditions are true:
+//
+// 1. the Protocol field is "HTTP", "HTTPS" or "TLS",
+// 2. the Hostname field is specified
+// 3. and the Port field matches
+//
+// As a special case, each group of collapsed Listeners may
+// contain up to one Listener that omits the Hostname field. Since
+// omitting the Hostname field captures all traffic directed to the
+// specified port, allowing more than one of these Listeners would
+// create either ambiguity or redundancy.
+//
+// Listeners that specify identical Hostname fields must not be
+// treated as collapsible. Put another way, each Hostname field in
+// a collapsible group must be unique in order to avoid ambiguity.
+//
+// If the GatewayClass collapses Listeners, the host name provided
+// in the incoming client request must be matched to a Listener to
+// find the correct set of Routes. The incoming host name must be matched
+// using the Hostname field for each listener in order of most to least
+// specific. That is, "Exact" matches must be processed before
+// "Domain" matches, which must be processed before Listeners that
+// omit the Hostname field.
+//
+// If the Gateway specifies multiple Listeners that have the same
+// Port value but are not collapsible, the GatewayClass must raise
+// a "PortConflict" condition on the Gateway.
 type Listener struct {
-	// Name is the listener's name and should be specified as an
-	// RFC 1035 DNS_LABEL [1]:
+	// Hostname specifies the virtual host name for protocol types
+	// that define this concept.
 	//
-	// [1] https://tools.ietf.org/html/rfc1035
+	// Incoming requests are matched against Hostname before processing
+	// HTTPRoute rules. For example, if the HTTP request header
+	// contains "Host: foo.example.com", a Listener with Hostname value
+	// "foo.example.com" will be an "Exact" match, but a Listener with
+	// Hostname values "example.com" or "bar.example.com" will not
+	// match. However these host names would match against a "Domain"
+	// match type with a value of "example.com".
 	//
-	// Each listener of a Gateway must have a unique name. Name is used
-	// for associating a listener in Gateway status.
+	// If Hostname is unspecified, the Gateway routes all traffic to
+	// the specified routes. This is equivalent to matching all host
+	// and domain names.
 	//
-	// Support: Core
+	// A Hostname match should be supplied for "HTTP" and "HTTPS"
+	// protocols, and may be supplied for "TLS" protocol. If no
+	// match is supplied, then this Listener must accept all client
```

Comment:
Can use more generic language here:

if hostname is specified, it will be matched according to the semantics of the protocol, describe behavior in the protocol enum to avoid duplication.

Hostname is optional? (you have should and may here)

to be accurate, the connection is accepted  but will result in 404ish result

Reply:
> to be accurate, the connection is accepted but will result in 404ish result

Can you clarify what text you are referring to here?

Reply:
To get the request at all, the connection will be accept()'ed, even if the TLS handshake is bad, host is wrong etc.
I'm referring to 

"if no match is supplied, then this Listener must accept all client *connections*"

Reply:
Oh I see. The new language is "accepts all client traffic", but this doesn't mean it proxies it, just that the listener gets the packets. It might do anything with them.